compliance_checker/cf/cf.py

```diff
@@ -2360,67 +2360,68 @@ def check_packed_data(self, ds):
         When data to be packed contains missing values the attributes that indicate missing values (_FillValue, valid_min,
         valid_max, valid_range) must be of the same data type as the packed data.
         """
-        
         ret_val = []
         reasoning = []
-        names = list(ds.dataset.variables.iterkeys())
         for name, var in ds.dataset.variables.iteritems():
-            if hasattr(var, 'add_offset') or hasattr(var, 'scale_factor'):
+            if hasattr(var, 'add_offset') and hasattr(var, 'scale_factor'):
```

Comment:
It is valid to have just one of these attributes defined, so we need to handle that case as well. For the purpose of this check, if one of them is not present, we assume it has the same dtype as the other.